apps/server/src/modules/tool/external-tool/uc/external-tool.uc.spec.ts

```diff
@@ -868,6 +957,93 @@ describe(ExternalToolUc.name, () => {
 				);
 			});
 		});
+
+		describe('when lti11 config is given and secret is not encrypted', () => {
+			const setupLTI = () => {
+				const { externalTool, toolId, mockLogoBase64 } = setupDefault();
+
+				const externalToolToUpdate: ExternalToolUpdate = {
+					...externalTool.getProps(),
+					name: 'newName',
+					url: undefined,
+				};
+				const updatedExternalTool: ExternalTool = externalToolFactory.build({
+					...externalTool.getProps(),
+					name: 'newName',
+					url: undefined,
+				});
+
+				const lti11ToolConfig: Lti11ToolConfig = lti11ToolConfigFactory.build();
+				externalToolToUpdate.config = lti11ToolConfig;
+
+				const currentTestTool = new ExternalTool({ ...externalToolToUpdate });
+				currentTestTool.config = lti11ToolConfigFactory.buildWithId({ secret: 'encryptedSecret' });
+
+				externalToolService.findById.mockResolvedValue(currentTestTool);
+				encryptionService.encrypt.mockReturnValue(lti11ToolConfig.secret);
```

Comment:
You are returning the unencrypted secret from the encrypt method. This way you can't test if the encrypted value is written back to the object correctly.